torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -393,8 +393,7 @@ def _local_post_state_dict_hook(
     shard_offset = flat_param.numel() * fsdp_state.rank
     valid_data_size = flat_param.numel() - flat_param._shard_numel_padded
     if valid_data_size > 0:
-        if flat_param._shard_numel_padded > 0:
-            flat_param = flat_param.narrow(0, 0, valid_data_size)
+        flat_param = flat_param.view(valid_data_size)
```

Comment:
nice, maybe add a comment saying this will make it return a tensor that can be properly serialized?